src/type_translator.ts

```diff
@@ -378,6 +378,26 @@ export class TypeTranslator {
     }
   }
 
+  /**
+   * Special-case translation of union types to work off the syntactical type representation
+   * (`ts.Node` object), as opposed to the semantic type representation (`ts.Type` object).
+   *
+   * Unlike with othe types, TypeScript expands aliased union types as they are added into another
+   * union. This means we cannot recover the original type nodes and names from the union type, and
```

Comment:
s/the union type/the ts.Type/ perhaps?  It confusing here because when you say "the union type" you mean something other than the 'UnionTypeNode' seen just below.

Reply:
Code removed, no longer applies.